### ISSUE TITLE
Question about command to produce a species and family pangenome database.

### ISSUE THREAD

**Issue by yuenchingshen** [dib-lab/2022-database-covers#9]
Should I use gtdb-rs214-k21.zip to run the command to get database for both genus, species and family?
Or use 'gtdb-rs214-k21.zip' to get database for genus, and run 'gtdb-rs214-k21.pangenomes.genus.zip' to get database to get species database, and then run 'gtdb-rs214-k21.pangenomes.species.zip' to get family database.

> ./pangenomeproject/make-pangenome-sketches.py gtdb-rs214-k21.zip -t gtdb-rs214.lineages.sqldb -o gtdb-rs214-k21.pangenomes.genus.zip -k 21 -r species 

> ./pangenomeproject/make-pangenome-sketches.py gtdb-rs214-k21.pangenomes.genus.zip -t gtdb-rs214.lineages.sqldb -o gtdb-rs214-k21.pangenomes.genus.zip -k 21 -r species

**Comment by yuenchingshen** [comment]
@ccbaumler hi Colton, I have a question on making the pangenome database.

**Comment by ccbaumler** [comment]
Considering the script is taking the existing minhash set for each lower rank and combining the sets, you should receive the same results with either database.

The set operation will only include unique items when merging sets. That said you could run both commands and check the results as well!

**Comment by yuenchingshen** [comment]
@ccbaumler Got it! Thank you.

**Comment by yuenchingshen** [comment]
@ccbaumler Hi, Colton
I run the command two times after our meeting, and it said the computer has trouble saving the file because there is a duplicate. But I also could not file the duplicate file...
>./pangenomeproject/make-pangenome-sketches.py gtdb-rs214-k21.zip -t gtdb-rs214.lineages.sqldb -o gtdb-rs214-k21.pangenomes.genus.zip -k 21 -r species


> ...84000 - saving
...85000 - saving
/home/yc22/miniforge3/envs/sourmash/lib/python3.12/zipfile/__init__.py:1598: UserWarning: Duplicate name: 'SOURMASH-MANIFEST.csv'
  return self._open_to_write(zinfo, force_zip64=force_zip64)
client_loop: send disconnect: Broken pipe

>(base) yc22@farm:~$ grep . -rli "SOURMASH-MANIFEST.csv"
grep: SOURMASH-MANIFEST.csv: No such file or directory

**Comment by ccbaumler** [comment]
@yuenchingshen Hey! I looked the `UserWarning` up in a search engine and found [this stackoverflow answer](https://stackoverflow.com/questions/67646030/python-zip-file-hit-userwarning-duplicate-name). 

TLDR: Your output is trying to overwrite an existing file of the same name and it is breaking the zip function.

You have four options:

1. change your `-o` filename to a unique name
2. delete your existing file of the same name
3. move your existing file
4. place the output file in a unique path

PS. Your command has an error in it.
>  ./pangenomeproject/make-pangenome-sketches.py gtdb-rs214-k21.zip -t gtdb-rs214.lineages.sqldb -o gtdb-rs214-k21.pangenomes.genus.zip -k 21 -r species

Your output name is designating this new pangenome database as a genus level database, but your rank argument is choosing species.

**Comment by yuenchingshen** [comment]
@ccbaumler 
ohhh I see
So I just need to change the command to
> ./pangenomeproject/make-pangenome-sketches.py gtdb-rs214-k21.zip -t gtdb-rs214.lineages.sqldb -o gtdb-rs214-k21.pangenomes.species.zip -k 21 -r species

> ./pangenomeproject/make-pangenome-sketches.py gtdb-rs214-k21.zip -t gtdb-rs214.lineages.sqldb -o gtdb-rs214-k21.pangenomes.family.zip -k 21 -r family

 thank you Colton.

**Comment by ctb** [comment]
> You have four options:
> 
>     1. change your `-o` filename to a unique name
> 
>     2. delete your existing file of the same name
> 
>     3. move your existing file
> 
>     4. place the output file in a unique path

you could also change the code to refuse to do anything if the file exists - we do that [here](https://github.com/sourmash-bio/sourmash/blob/latest/src/sourmash/sig/__main__.py#L1526), in sourmash sig collect.
